Port android app 2025.4-beta2 to main (#8222)
mullvad/mullvadvpn-app: android/CHANGELOG.md

  is drastically smaller.

## [android/2025.4-beta2] - 2025-05-20
### Changed

> - Improve error message when the app can not connect due to an invalid port.

Reviewer: ```suggestion
- Improve error message when the app cannot connect due to an invalid port.
```
